keps/sig-apps/2307-job-tracking-without-lingering-pods/README.md

```diff
@@ -570,20 +653,21 @@ The flow was completed successfully with all the stated verifications.
 
 #### What are other known failure modes?
   
-  TBD from user feedback. No know failures modes so far.
-  
-  <!--
-  For each of them, fill in the following information by copying the below template:
-  - [Failure mode brief description]
-    - Detection: How can it be detected via metrics? Stated another way:
-      how can an operator troubleshoot without logging into a master or worker node?
-    - Mitigations: What can be done to stop the bleeding, especially for already
-      running user workloads?
-    - Diagnostics: What are the useful log messages and their required logging
-      levels that could help debug the issue?
-      Not required until feature graduated to beta.
-    - Testing: Are there any tests for failure mode? If not, describe why.
-  -->
+  - Terminated pods are stuck with finalizers
+    - Detection:
+      - Before 1.26: Observe the behavior in pods.
+      - After 1.26: Based on metric `job_terminated_pod_tracking_finalizer`
+    - Mitigations:
+      Before 1.26, disable `JobTrackingWithFinalizers`.
+    - Diagnostics:
+      The job controller reports errors updating the Job status and/or patching
+      Pods.
+      There were some bugs that would cause this (examples:
+      [#109485](https://github.com/kubernetes/kubernetes/issues/109485),
+      [#111646](https://github.com/kubernetes/kubernetes/pull/111646)).
+      In newer versions, this can still happen if there is a buggy webhook
+      that prevents pod updates to remove finalizers.
+    - Testing: Discovered bugs are covered by unit and integration tests.
```

Comment:
I'd add here the problem you described at the beginning, ie. the one that we recommend folks running with `JobTrackingWithFinalizers` in 1.25 and 1.26 to ensure all jobs with `batch.kubernetes.io/job-completion` are completed prior to upgrade to 1.27. I doubt we'll be updating the KEP then, so let's add it right away.

Reply:
On top of that, I'd suggest writing a blog post for 1.26 release describing this and recommendations.

Reply:
Good ideas. Added here and in the graduation criteria.